Hotfix conda

tdcwilliams opened the pull request:
- For some reason conda wasn't able to install any combination of packages that I tried inside docker.
  - By making a virtual environment inside docker they can be installed now.
  - Added `conda activate pynextsim` to `/root/.bashrc` so it is automatically activated for an interactive container.
- To run a command in docker, from outside docker, it is slightly more complicated as you have to activate pynextsim before running. However it is not too bad: see
https://github.com/nansencenter/nextsim-env/blob/master/machines/fram.nersc.no-tim/run_cmd_in_docker.sh, where the `$DOCKER_RUN` command there is eg
```
export DOCKER_RUN="docker run --rm \
    -e NEXTSIMDIR=/nextsim \
    -e INPUT_OBS_DATA_DIR=/Data/sim/data \
    -e INPUT_FORECAST_DATA_DIR=/Data/nextsimf/data \
    -e NEXTSIM_TEST_DATA_DIR=/nextsim-tools/example_data \
    -e OUTPUT_FORECAST_DATA_DIR=/docker_io/forecasts \
    -e TEST_FORECAST_DATA_DIR=$TEST_FORECAST_DATA_DIR \
    -e PYTHONUNBUFFERED=1 \
    -v /Data/sim/:/Data/sim/ \
    -v /Data/nextsimf/:/Data/nextsimf/ \
    -v /mnt/10.11.12.231/sim/:/mnt/10.11.12.231/sim/ \
    -v /mnt/10.11.12.232/nextsimf/:/mnt/10.11.12.232/nextsimf/ \
    -v $NEXTSIMDIR:/nextsim/ \
    -v $NEXTSIM_DATA_DIR:/data/ \
    -v $NEXTSIM_MESH_DIR:/mesh/ \
    -v $NEXTSIMTOOLS_ROOT_DIR:/nextsim-tools/ \
    -v $NEXTSIMF_ROOT_DIR:/nextsimf/ \
    -v $SWARP_ROUTINES:/SWARP-routines \
    -v $NEXTSIM_ENV_ROOT_DIR:/nextsim-env \
    -v $OUTPUT_FORECAST_DATA_DIR:/docker_io/forecasts/ \
    -v $MY_ENV_DIR:/root/my_env_dir/ \
    -v $HOME/.ssh:/root/.ssh/ \
    --security-opt seccomp=unconfined \
    --network=host
```
to mount the required directories and set environment variables.

Other changes:
- install lftp, ncftp
- remove SIM_USER stuff as not needed anymore

akorosov commented:
Tim, I simplified the Dockerfile: 
I guess you dont need two stage build - removed.
No magic with __conda_setup is needed. Just used COND_DEFAULT_ENV.
Replaced requirements.txt with environment.yml
Removed some of the packages that are probably not needed yet. 

Can you test it?

tdcwilliams commented:
thanks @akorosov 
I'll try it out. I'll add back pandas, which I am using right now to keep track of the status of the files that have been uploaded (uploaded, ingested, delete_requested, deleted...)

tdcwilliams commented:
hi @akorosov - nice improvements - it seems to work after I added basemap back (haven't quite got rid of it yet); also added pandas again. Could you push the latest version to the dockerhub please?
We also need to update the nextsim-tools Dockerfile after that - there is a PR for that too (works with these changes)
